### PR TITLE
port c2rust-transpile to syn, disable c2rust-refactor, llvm_asm! -> asm!

### PR DESCRIPTION
**Pull request by fw-immunant**
This makes it possible to build c2rust with a current nightly (and one as old as a few months ago probably works, but I'm not sure of the exact compat range).

The caveats:
- The c2rust-refactor tool has not been ported, as it depends not just on rustc's libsyntax but also on lots of other rustc internals, and the point of this is to *separate* the most useful part of c2rust from being tied to a specific rustc version. But don't give up all hope for cleaning up your unsafe c2rust-generated Rust code--good things may show up on this front in the future.
- Comments are no longer preserved in translated code. This is not a permanent change (expect a follow-up PR to reintroduce comments in most AST positions), but comment translation has to be implemented in a substantially different way when using syn+prettyplease for AST generation and pretty-printing, and this is still WIP.

### CODE REVIEW

**Comment by XVilka** [comment]
Will it be possible to generate Rust code using this port, and then use the c2rust-refactor from older commits of the c2rust?

**Comment by fw-immunant** [comment]
> Will it be possible to generate Rust code using this port, and then use the c2rust-refactor from older commits of the c2rust?

Yes, the rust code generated by the new c2rust-transpile should still work with the old c2rust-refactor, though that workflow isn't something with test coverage, so your mileage may vary as c2rust-transpile evolves with language and compiler changes.

**Comment by nacaclanga** [comment]
I assume this compleately breaks c2rust-refactor. Maybe we can rewrite c2rust-refactor and c2rust transpiler into two entierly different programms, that can be installed individually with c2rust itself, just checking for them to exist and call them. We can do so by adding a feature to c2rust-ast-printer and c2rust-ast-builder that switches from syn to a libsyntax version for now. This allows for people to still be able to use the unstable old refactor tool, if they want to.

**Comment by rinon** [comment]
> I assume this compleately breaks c2rust-refactor. Maybe we can rewrite c2rust-refactor and c2rust transpiler into two entierly different programms, that can be installed individually with c2rust itself, just checking for them to exist and call them. We can do so by adding a feature to c2rust-ast-printer and c2rust-ast-builder that switches from syn to a libsyntax version for now. This allows for people to still be able to use the unstable old refactor tool, if they want to.

Maintaining both with a feature flag seems infeasible, given the extent of these changes. Building c2rust-refactor is still easy, however, with the skew in nightly versions between the transpiler and the refactoring tool, it's more complicated than just a feature flag. You're welcome to still build and use the refactoring tool, either by modifying the cargo configuration or just pinning to an old commit, but I don't see a good way to support both concurrently for the moment.

Keep in mind that code from the new transpiler will probably not parse with the old nightly (and vice versa, because we use too many nightly features in the transpiled output code).

**Review by rinon — APPROVED** [review]

File: c2rust-transpile/src/translator/assembly.rs [hunk 1]
```diff
@@ -2,6 +2,330 @@
...
+}
+
+fn translate_modifier(modifier: char, arch: &str) -> Option<char> {
+    Some(match arch {
+        "x86" => match modifier {
+            'k' => 'e',
+            'q' => 'r',
+            'b' => 'l',
+            'h' => 'h',
+            'w' => 'x',
```
[the remark this answers is not in the document]
Comment (fw-immunant) [reply c2rust-transpile/src/translator/assembly.rs:131]: Done.

**Pull request merged**